### ISSUE TITLE
ModuleNotFoundError: No module named 'vaex.astro'/'vaex.viz'

### ISSUE THREAD

**Issue by ymdt315** [vaexio/vaex#464]
I just tried to install vaex and execute test code(on AWS EC2, CentOS).
but ModuleNotFoundError is returned.

[install]
```bash
pip --no-cache-dir install -I --pre vaex

Collecting vaex
  Downloading https://files.pythonhosted.org/packages/bb/e6/0f7f6fed83c6b9698e86dbdde00cb2a222569822239b023f4aa83ff1cfac/vaex-2.3.0.tar.gz
Collecting vaex-core<2,>=1.2.0
  Downloading https://files.pythonhosted.org/packages/8a/38/f2f67c1713a1be7341815cc8ad6200354645d2855e380ec7144816b87352/vaex_core-1.2.0-cp36-cp36m-manylinux1_x86_64.whl (2.2MB)
     |????????????????????????????????| 2.2MB 5.9MB/s
.......
.......
Collecting jmespath<1.0.0,>=0.7.1
  Downloading https://files.pythonhosted.org/packages/83/94/7179c3832a6d45b266ddb2aac329e101367fbdb11f425f13771d27f225bb/jmespath-0.9.4-py2.py3-none-any.whl
Collecting docutils<0.16,>=0.10
  Downloading https://files.pythonhosted.org/packages/22/cd/a6aa959dca619918ccb55023b4cb151949c64d4d5d55b3f4ffd7eee0c6e8/docutils-0.15.2-py3-none-any.whl (547kB)
     |????????????????????????????????| 552kB 72.8MB/s
ERROR: awscli 1.16.165 has requirement botocore==1.12.155, but you'll have botocore 1.13.21 which is incompatible.
ERROR: awscli 1.16.165 has requirement PyYAML<=3.13,>=3.10, but you'll have pyyaml 5.1.2 which is incompatible.
ERROR: botocore 1.13.21 has requirement python-dateutil<2.8.1,>=2.1; python_version >= "2.7", but you'll have python-dateutil 2.8.1 which is incompatible.
Installing collected packages: pytz, six, python-dateutil, numpy, pandas, pyyaml, future, certifi, chardet, idna, urllib3, requests, cloudpickle, tabulate, aplus, astropy, psutil, python-utils, progressbar2, vaex-core, cycler, setuptools, kiwisolver, pyparsing, matplotlib, pillow, vaex-viz, tornado, cachetools, vaex-server, h5py, docutils, jmespath, botocore, s3transfer, boto3, s3fs, vaex-hdf5, vaex-astro, pyarrow, vaex-arrow, vaex
    Running setup.py install for pyyaml ... done
    Running setup.py install for future ... done
    Running setup.py install for tabulate ... done
    Running setup.py install for aplus ... done
    Running setup.py install for psutil ... done
    Running setup.py install for vaex-viz ... done
    Running setup.py install for tornado ... done
    Running setup.py install for vaex-server ... done
    Running setup.py install for s3fs ... done
    Running setup.py install for vaex-hdf5 ... done
    Running setup.py install for vaex-astro ... done
    Running setup.py install for vaex-arrow ... done
    Running setup.py install for vaex ... done
Successfully installed aplus-0.11.0 astropy-4.0rc1 boto3-1.10.21 botocore-1.13.21 cachetools-3.1.1 certifi-2019.9.11 chardet-3.0.4 cloudpickle-1.2.2 cycler-0.10.0 docutils-0.15.2 future-0.18.2 h5py-2.10.0 idna-2.8 jmespath-0.9.4 kiwisolver-1.1.0 matplotlib-3.2.0rc1 numpy-1.17.4 pandas-0.25.3 pillow-6.2.1 progressbar2-3.47.0 psutil-5.6.5 pyarrow-0.15.1 pyparsing-2.4.5 python-dateutil-2.8.1 python-utils-2.3.0 pytz-2019.3 pyyaml-5.1.2 requests-2.22.0 s3fs-0.2.2 s3transfer-0.2.1 setuptools-41.6.0 six-1.13.0 tabulate-0.8.6 tornado-6.0.3 urllib3-1.25.7 vaex-2.3.0 vaex-arrow-0.4.1 vaex-astro-0.6.1 vaex-core-1.2.0 vaex-hdf5-0.5.5 vaex-server-0.2.1 vaex-viz-0.3.8
```

[execute vaex]
```bash
python -c "import vaex
vx = vaex.open('/home/user001/test.csv')"

ERROR:MainThread:vaex:issue loading plot
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/vaex/__init__.py", line 720, in <module>
    add_namespace = entry.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2443, in load
    return self.resolve()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2449, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
ModuleNotFoundError: No module named 'vaex.viz'
ERROR:MainThread:vaex:issue loading astro
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/vaex/__init__.py", line 720, in <module>
    add_namespace = entry.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2443, in load
    return self.resolve()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2449, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
ModuleNotFoundError: No module named 'vaex.astro'`
```

I don't know why vaex can't find the module I installed.

**Comment by JovanVeljanoski** [comment]
Hi @ymdt315 

Can you perhaps try:
`pip install vaex==2`
or 
`pip install vaex==2.3.0`

Cheers, 
Jovan.

**Comment by ymdt315** [comment]
Hi @JovanVeljanoski, thank you for quick reply!

I tried pip install vaex==2, but another error is occured...
it seems to be version conflict

[install]
```bash
pip install vaex==2

Collecting vaex==2
  Downloading https://files.pythonhosted.org/packages/5e/2e/18e84946f041f48137d819c115eede0a2eeabd586f6740fe99743cb024c0/vaex-2.0.0.tar.gz
Collecting vaex-core<0.10,>=0.9.0
  Downloading https://files.pythonhosted.org/packages/03/83/c4cea06ae1e8e4e85f83db4c7ced4e2d99651b53e762f8f269525b3d94ee/vaex_core-0.9.2-cp36-cp36m-manylinux1_x86_64.whl (1.7MB)
     |????????????????????????????????| 1.7MB 6.6MB/s
Requirement already satisfied: vaex-viz<0.4,>=0.3.3 in /usr/lib64/python3.6/site-packages (from vaex==2) (0.3.8)
Requirement already satisfied: vaex-server<0.3,>=0.2.1 in /usr/lib/python3.6/site-packages (from vaex==2) (0.2.1)
Requirement already satisfied: vaex-hdf5<0.6,>=0.5.2 in /usr/lib64/python3.6/site-packages (from vaex==2) (0.5.5)
Collecting vaex-astro<0.6,>=0.5.0
  Downloading https://files.pythonhosted.org/packages/11/9f/3730aa70d6c4b48c68cb62c8a48618e71eabf60a0b09fa1cad0f4bc405e9/vaex-astro-0.5.0.tar.gz
Collecting vaex-arrow<0.4,>=0.3.4
  Using cached https://files.pythonhosted.org/packages/9e/4b/62848ca13b2b39e4c8d387a579c13b96f4b0567cf8f0241a507ec594c0f9/vaex-arrow-0.3.5.tar.gz
Requirement already satisfied: astropy>=2 in /usr/lib64/python3.6/site-packages (from vaex-core<0.10,>=0.9.0->vaex==2) (4.0rc1)
Requirement already satisfied: progressbar2 in /usr/lib/python3.6/site-packages (from vaex-core<0.10,>=0.9.0->vaex==2) (3.47.0)
.........
.........
Requirement already satisfied: botocore>=1.12.91 in /usr/lib/python3.6/site-packages (from s3fs<0.3->vaex-hdf5<0.6,>=0.5.2->vaex==2) (1.13.21)
Requirement already satisfied: setuptools in /usr/lib/python3.6/site-packages (from kiwisolver>=1.0.1->matplotlib>=1.3.1->vaex-viz<0.4,>=0.3.3->vaex==2) (41.6.0)
Requirement already satisfied: s3transfer<0.3.0,>=0.2.0 in /root/.local/lib/python3.6/site-packages (from boto3>=1.9.91->s3fs<0.3->vaex-hdf5<0.6,>=0.5.2->vaex==2) (0.2.0)
Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /root/.local/lib/python3.6/site-packages (from boto3>=1.9.91->s3fs<0.3->vaex-hdf5<0.6,>=0.5.2->vaex==2) (0.9.4)
Requirement already satisfied: docutils<0.16,>=0.10 in /root/.local/lib/python3.6/site-packages (from botocore>=1.12.91->s3fs<0.3->vaex-hdf5<0.6,>=0.5.2->vaex==2) (0.14)
ERROR: vaex-viz 0.3.8 has requirement vaex-core<2,>=1.0.0, but you'll have vaex-core 0.9.2 which is incompatible.
ERROR: vaex-hdf5 0.5.5 has requirement vaex-core<2,>=1.0.0, but you'll have vaex-core 0.9.2 which is incompatible.
ERROR: vaex-ml 0.6.1 has requirement vaex-core<2,>=1.0.0, but you'll have vaex-core 0.9.2 which is incompatible.
ERROR: vaex-jupyter 0.4.0 has requirement vaex-core<2,>=1.0.0, but you'll have vaex-core 0.9.2 which is incompatible.
Installing collected packages: vaex-core, vaex-astro, vaex-arrow, vaex
  Found existing installation: vaex-core 1.2.0
    Uninstalling vaex-core-1.2.0:
      Successfully uninstalled vaex-core-1.2.0
  Found existing installation: vaex-astro 0.6.1
    Can't uninstall 'vaex-astro'. No files were found to uninstall.
    Running setup.py install for vaex-astro ... done
  Found existing installation: vaex-arrow 0.4.1
    Uninstalling vaex-arrow-0.4.1:
      Successfully uninstalled vaex-arrow-0.4.1
    Running setup.py install for vaex-arrow ... done
  Found existing installation: vaex 2.3.0
    Uninstalling vaex-2.3.0:
      Successfully uninstalled vaex-2.3.0
    Running setup.py install for vaex ... done
Successfully installed vaex-2.0.0 vaex-arrow-0.4.1 vaex-astro-0.6.1 vaex-core-0.9.2
```


[execute]
```bash
python -c "import vaex
vx = vaex.open('/home/user001/test.csv')"


Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/vaex/__init__.py", line 648, in <module>
    add_namespace = entry.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2442, in load
    self.require(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2465, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.VersionConflict: (vaex-core 0.9.2 (/usr/lib64/python3.6/site-packages), Requirement.parse('vaex-core<2,>=1.0.0'))
ERROR:MainThread:vaex:issue loading astro
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/vaex/__init__.py", line 648, in <module>
    add_namespace = entry.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2442, in load
    self.require(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2465, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.VersionConflict: (vaex-core 0.9.2 (/usr/lib64/python3.6/site-packages), Requirement.parse('vaex-core<2,>=1.0.0'))
```


[ref]
```bash
cat /home/user001/test.csv

1,2,3
```

**Comment by ymdt315** [comment]
to avoid version conflict, I used 'pip install vaex-core==1'.
but ModuleNotFoundError: No module named 'vaex.astro'/'vaex.viz' returned again.

```bash
pip install vaex-core==1

Collecting vaex-core==1
  Using cached https://files.pythonhosted.org/packages/91/33/ab78717cb9a2347c3881ea1f668c5a9855840676bc6689a89cee34bbea28/vaex_core-1.0.0-cp36-cp36m-manylinux1_x86_64.whl
Requirement already satisfied: six in /usr/lib/python3.6/site-packages (from vaex-core==1) (1.13.0)
Requirement already satisfied: astropy>=2 in /usr/lib64/python3.6/site-packages (from vaex-core==1) (4.0rc1)
Requirement already satisfied: aplus in /usr/lib/python3.6/site-packages (from vaex-core==1) (0.11.0)
............
............
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python3.6/site-packages (from requests->vaex-core==1) (3.0.4)
Requirement already satisfied: python-utils>=2.3.0 in /usr/lib/python3.6/site-packages (from progressbar2->vaex-core==1) (2.3.0)
ERROR: vaex 2.0.0 has requirement vaex-astro<0.6,>=0.5.0, but you'll have vaex-astro 0.6.1 which is incompatible.
ERROR: vaex 2.0.0 has requirement vaex-core<0.10,>=0.9.0, but you'll have vaex-core 1.0.0 which is incompatible.
Installing collected packages: vaex-core
  Found existing installation: vaex-core 0.9.2
    Uninstalling vaex-core-0.9.2:
      Successfully uninstalled vaex-core-0.9.2
Successfully installed vaex-core-1.0.0
```

```bash
python -c "import vaex
vx = vaex.open('/home/user001/test.csv')"

ERROR:MainThread:vaex:issue loading plot
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/vaex/__init__.py", line 698, in <module>
    add_namespace = entry.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2443, in load
    return self.resolve()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2449, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
ModuleNotFoundError: No module named 'vaex.viz'
ERROR:MainThread:vaex:issue loading astro
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/vaex/__init__.py", line 698, in <module>
    add_namespace = entry.load()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2443, in load
    return self.resolve()
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2449, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
ModuleNotFoundError: No module named 'vaex.astro'
```

**Comment by JovanVeljanoski** [comment]
Weird.. pip is doing some weird caching, sometimes it causes problems with versions.. 

Could you try the following please:
- option 1:
Uninstall vaex. pip uninstall vaex may not be enough.. you may need to uninstall each package in the individually (pip uninstall vaex-core, pip uninstall vaex-viz arrow etc..).
Once that is done do:
`pip install vaex==2.3.0`.

- option 2:
`pip install -I vaex==2.3.0` to force pip to reinstall stuff, hopefully this will fix the dependencies. 

- option 3:
Uninstall vaex like in option 1. Then do `pip install vaex-core==1.2.0`.
After that you can do `pip install vaex==2.3.0`. 

In all cases, if the installation goes as planned, when you do `pip search vaex`, you should get that vaex-core is 1.2.0 (latest). 

- option 4:
Perhaps, if this is possible for you, install python/vaex through the conda package manager. Since it solves dependencies quite nicely, you will not have such problems as with pip. To do that, once you have conda set up do:
`conda install -c conda-forge vaex`

I hope this helps, 
Jovan.

**Comment by maartenbreddels** [comment]
Another idea would be to upgrade pip and/or setuptools, we've seen these issues in the past a lot, so maybe that is the issue. We'd love to hear back from you on this!

**Comment by ymdt315** [comment]
Thank you for your kindness,  @JovanVeljanoski @maartenbreddels

I tried option one by one, but finally failed...


1. update pip

```bash
python -m pip install --upgrade pip

Requirement already up-to-date: pip in /usr/lib/python3.6/site-packages (19.3.1)
```

2. update setuptools

```bash
pip install setuptools --upgrade

Requirement already up-to-date: setuptools in /usr/lib/python3.6/site-packages (41.6.0)
```


3. execute test command, but failed(same error message)


4. re-install vaex

```bash

pip list
.......
vaex               2.0.0
vaex-arrow         0.3.5
vaex-astro         0.6.1
vaex-core          1.0.0
vaex-distributed   0.3.0
vaex-hdf5          0.5.5
vaex-jupyter       0.4.0
vaex-ml            0.6.1
vaex-server        0.2.1
vaex-viz           0.3.8
.......

////////////////////////////////////////////////////////

pip uninstall vaex vaex-arrow vaex-astro vaex-core vaex-distributed vaex-hdf5 vaex-jupyter vaex-ml vaex-server vaex-viz

Uninstalling vaex-2.0.0:
  Would remove:
    /usr/lib/python3.6/site-packages/vaex-2.0.0-py3.6.egg-info
    /usr/lib/python3.6/site-packages/vaex/meta/_version.py
Proceed (y/n)? y
  Successfully uninstalled vaex-2.0.0
Uninstalling vaex-arrow-0.3.5:
  Would remove:
    /usr/lib/python3.6/site-packages/vaex_arrow-0.3.5-py3.6.egg-info
    /usr/lib/python3.6/site-packages/vaex_arrow/*
Proceed (y/n)? y
  Successfully uninstalled vaex-arrow-0.3.5
Can't uninstall 'vaex-astro'. No files were found to uninstall.
Uninstalling vaex-core-1.0.0:
  Would remove:
    /usr/bin/vaex
    /usr/bin/vaexgui
    /usr/lib64/python3.6/site-packages/vaex/*
    /usr/lib64/python3.6/site-packages/vaex_core-1.0.0.dist-info/*
Proceed (y/n)? y
  Successfully uninstalled vaex-core-1.0.0
Uninstalling vaex-distributed-0.3.0:
  Would remove:
    /usr/lib/python3.6/site-packages/vaex/distributed/*
    /usr/lib/python3.6/site-packages/vaex_distributed-0.3.0-py3.6.egg-info
Proceed (y/n)? y
  Successfully uninstalled vaex-distributed-0.3.0
Uninstalling vaex-hdf5-0.5.5:
  Would remove:
    /usr/lib64/python3.6/site-packages/vaex_hdf5-0.5.5-py3.6.egg-info
Proceed (y/n)? y
  Successfully uninstalled vaex-hdf5-0.5.5
Uninstalling vaex-jupyter-0.4.0:
  Would remove:
    /usr/lib/python3.6/site-packages/vaex/jupyter/*
    /usr/lib/python3.6/site-packages/vaex_jupyter-0.4.0-py3.6.egg-info
Proceed (y/n)? y
  Successfully uninstalled vaex-jupyter-0.4.0
Uninstalling vaex-ml-0.6.1:
  Would remove:
    /usr/lib/python3.6/site-packages/vaex/ml/*
    /usr/lib/python3.6/site-packages/vaex_ml-0.6.1-py3.6.egg-info
Proceed (y/n)? y
  Successfully uninstalled vaex-ml-0.6.1
Uninstalling vaex-server-0.2.1:
  Would remove:
    /usr/lib/python3.6/site-packages/vaex/server/*
    /usr/lib/python3.6/site-packages/vaex_server-0.2.1-py3.6.egg-info
Proceed (y/n)? y
  Successfully uninstalled vaex-server-0.2.1
Can't uninstall 'vaex-viz'. No files were found to uninstall.

////////////////////////////////////////////////////////

pip install vaex==2.3.0

Collecting vaex==2.3.0
  Using cached https://files.pythonhosted.org/packages/bb/e6/0f7f6fed83c6b9698e86dbdde00cb2a222569822239b023f4aa83ff1cfac/vaex-2.3.0.tar.gz
Collecting vaex-core<2,>=1.2.0
  Using cached https://files.pythonhosted.org/packages/8a/38/f2f67c1713a1be7341815cc8ad6200354645d2855e380ec7144816b87352/vaex_core-1.2.0-cp36-cp36m-manylinux1_x86_64.whl
Requirement already satisfied: vaex-viz<0.4,>=0.3.8 in /usr/lib64/python3.6/site-packages (from vaex==2.3.0) (0.3.8)
Collecting vaex-server<0.3,>=0.2.1
  Using cached https://files.pythonhosted.org/packages/5b/b2/77a32cc6177e24c48af19d7179722574a51aabf4964386585f0ce30f718f/vaex-server-0.2.1.tar.gz
Requirement already satisfied: vaex-hdf5<0.6,>=0.5.5 in /usr/lib/python3.6/site-packages (from vaex==2.3.0) (0.5.5)
Requirement already satisfied: vaex-astro<0.7,>=0.6.1 in /usr/lib64/python3.6/site-packages (from vaex==2.3.0) (0.6.1)
Collecting vaex-arrow<0.5,>=0.4.1
  Using cached https://files.pythonhosted.org/packages/eb/66/8bb2128d6d55b1682be68f8320c4ded43996f23d43b561b7b0f3e988ffc6/vaex-arrow-0.4.1.tar.gz
Requirement already satisfied: astropy>=2 in /usr/lib64/python3.6/site-packages (from vaex-core<2,>=1.2.0->vaex==2.3.0) (4.0rc1)
Requirement already satisfied: aplus in /usr/lib/python3.6/site-packages (from vaex-core<2,>=1.2.0->vaex==2.3.0) (0.11.0)
.......
.......
Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in ./.local/lib/python3.6/site-packages (from boto3>=1.9.91->s3fs<0.3->vaex-hdf5<0.6,>=0.5.5->vaex==2.3.0) (0.9.4)
Requirement already satisfied: docutils<0.16,>=0.10 in ./.local/lib/python3.6/site-packages (from botocore>=1.12.91->s3fs<0.3->vaex-hdf5<0.6,>=0.5.5->vaex==2.3.0) (0.14)
Installing collected packages: vaex-core, vaex-server, vaex-arrow, vaex
    Running setup.py install for vaex-server ... done
    Running setup.py install for vaex-arrow ... done
    Running setup.py install for vaex ... done
Successfully installed vaex-2.3.0 vaex-arrow-0.4.1 vaex-core-1.2.0 vaex-server-0.2.1
```

5. execute test command, but failed(same error message)


6. use option 'pip install -I vaex==2.3.0'

```bash
Collecting vaex==2.3.0
  Using cached https://files.pythonhosted.org/packages/bb/e6/0f7f6fed83c6b9698e86dbdde00cb2a222569822239b023f4aa83ff1cfac/vaex-2.3.0.tar.gz
Collecting vaex-core<2,>=1.2.0
  Using cached https://files.pythonhosted.org/packages/8a/38/f2f67c1713a1be7341815cc8ad6200354645d2855e380ec7144816b87352/vaex_core-1.2.0-cp36-cp36m-manylinux1_x86_64.whl
Collecting vaex-viz<0.4,>=0.3.8
  Using cached https://files.pythonhosted.org/packages/4a/6c/301aac6cef5d262eeffdd286fff55b63f9e7dcff51825a236b3f7ee5e9fe/vaex-viz-0.3.8.tar.gz
Collecting vaex-server<0.3,>=0.2.1
......
Successfully installed aplus-0.11.0 astropy-4.0rc1 boto3-1.10.25 botocore-1.13.25 cachetools-3.1.1 certifi-2019.9.11 chardet-3.0.4 cloudpickle-1.2.2 cycler-0.10.0 docutils-0.15.2 future-0.18.2 h5py-2.10.0 idna-2.8 jmespath-0.9.4 kiwisolver-1.1.0 matplotlib-3.2.0rc1 numpy-1.17.4 pandas-0.25.3 pillow-6.2.1 progressbar2-3.47.0 psutil-5.6.5 pyarrow-0.15.1 pyparsing-2.4.5 python-dateutil-2.8.1 python-utils-2.3.0 pytz-2019.3 pyyaml-5.1.2 requests-2.22.0 s3fs-0.2.2 s3transfer-0.2.1 setuptools-41.6.0 six-1.13.0 tabulate-0.8.6 tornado-6.0.3 urllib3-1.25.7 vaex-2.3.0 vaex-arrow-0.4.1 vaex-astro-0.6.1 vaex-core-1.2.0 vaex-hdf5-0.5.5 vaex-server-0.2.1 vaex-viz-0.3.8
```

7. execute test command, but failed(same error message)


8. install vaex by using conda

9. execute test command, but failed(pyarrow conflict)

```bash
ERROR:MainThread:vaex:issue loading reader
Traceback (most recent call last):
  File "/root/anaconda3/lib/python3.7/site-packages/vaex/__init__.py", line 720, in <module>
    add_namespace = entry.load()
  File "/root/anaconda3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2433, in load
    self.require(*args, **kwargs)
  File "/root/anaconda3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2456, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/root/anaconda3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.VersionConflict: (pyarrow 0.11.1 (/root/anaconda3/lib/python3.7/site-packages), Requirement.parse('pyarrow>=0.15'))
```

10. to avoid conflict, try to upload pyarrow

```bash

conda install -c conda-forge pyarrow
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /root/anaconda3

  added / updated specs:
    - pyarrow


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2019.9.11          |           py37_0         147 KB  conda-forge
    conda-4.7.12               |           py37_0         3.0 MB  conda-forge
    ------------------------------------------------------------
                                           Total:         3.1 MB

The following packages will be SUPERSEDED by a higher-priority channel:

  ca-certificates    pkgs/main::ca-certificates-2019.10.16~ --> conda-forge::ca-certificates-2019.9.11-hecc5488_0
  certifi                                         pkgs/main --> conda-forge
  conda                                           pkgs/main --> conda-forge
  openssl              pkgs/main::openssl-1.1.1d-h7b6447c_3 --> conda-forge::openssl-1.1.1d-h516909a_0


Proceed ([y]/n)? y


Downloading and Extracting Packages
certifi-2019.9.11    | 147 KB    | ################################################################################################################################## | 100%
conda-4.7.12         | 3.0 MB    | ################################################################################################################################## | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done

```

11. execute test command, but failed(pyarrow conflict)

**Comment by JovanVeljanoski** [comment]
Hi, 

Since you are using conda.. you could also create a new environment, and try to install vaex though conda forge. Something like this:

```
conda create --name vaex_env
source activate vaex_enc
```
Then simply install vaex through conda(forge). I find this a bit more robust than pip, since conda will try to do a better job with all the dependencies etc:

`conda install vaex -c conda-forge`

This will install all necessary dependencies as well. 

If you can, please try this out and let us know. 

Cheers, 
Jovan.

**Comment by ymdt315** [comment]
thank you, @JovanVeljanoski 

It works!
I'll use this setting.
thank you for helping!

**Comment by FarahAmirahJ** [comment]
I have problem installing vaex
then I saw this thread. 
I try to install vaex by 
1) create new environment
2) using conda install -c conda-forge vaex

the problem comes when I try to import vaex.
it says, ModuleNotFoundError: No module named 'frozendict'
*after using conda install -c conda-forge vaex, the list for frozendict is 2.3.0

so I try to pip install frozendict
and try again, the output still the same:
ModuleNotFoundError: No module named 'frozendict'
![frozen](https://user-images.githubusercontent.com/102510231/164844335-7f9ce442-adbc-4611-a3c0-7b16f8de3800.jpg)

**Comment by maartenbreddels** [comment]
I have no idea how that is possible, frozendict is simply a dependency of vaex, and since you use conda, it should be installed. I'm assuming your conda installation is messed up, and maybe you want to start from scratch?
